pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentReplicator.java

```diff
@@ -704,10 +704,19 @@ public Optional<DispatchRateLimiter> getRateLimiter() {
     }
 
     @Override
-    public void initializeDispatchRateLimiterIfNeeded() {
+    public boolean initializeDispatchRateLimiterIfNeeded() {
         if (!dispatchRateLimiter.isPresent()
             && DispatchRateLimiter.isDispatchRateEnabled(topic.getReplicatorDispatchRate())) {
             this.dispatchRateLimiter = Optional.of(new DispatchRateLimiter(topic, Type.REPLICATOR));
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void updateRateLimiter() {
```

Comment:
Is it possible that this method will be called concurrently?

Reply:
Yes, I have updated the code to synchronize `initializeDispatchRateLimiterIfNeeded` and `updateRateLimiter`. PTAL again @Jason918 ~